### PR TITLE
Fix account tools Finish setup button: point to Stripe KYC flow instead of Stripe Express

### LINKED ISSUE #8585: The Account Tools Finish setup button should point to KYC flow
### Describe the bug

The "Finish setup" button in the Account Tools subsection of the Account Details section should link to the KYC flow for all accounts that didn't complete KYC (initial KYC for POed accounts), regardless if they are live or sandboxed accounts.

Furthermore, the "Finish setup" button should not be visible if the account has completed KYC (details_submitted).

Right now, the "Finish setup" button links to the Stripe Express dashboard, just like the "Edit details" link does.

Screenshot 2024-04-05 at 14 28 57

### To Reproduce

1. Onboard a PO account
2. Click on 'Finish setup' in the Account Tools sub-section
3. You end up in the Stripe Express dashboard

### Actual behavior
The Stripe Express dashboard is opened:

### Expected behavior
The Stripe KYC flow should be opened:

### Additional context
https://github.com/Automattic/woocommerce-payments/pull/7914

### PR DESCRIPTION
**Pull request by oaratovskyi**
Fixes #8585

#### Changes proposed in this Pull Request

* Fix Account Tools -> Finish setup button behaviour: redirect the merchant to KYC flow if they haven't completed it (instead of Stripe Express). Don't show up the button for account that completed the KYC. Reset button still can be shown for sandbox account that completed the KYC.

#### Testing instructions

* Checkout the PR branch (`fix/8585-account-tools-finish-setup-button-point-to-kyc-flow`) on your local installation and build the client assets with `npm run build:client`
* Onboard new account
* Return early from the onboarding, you should see the account tools with Finish Setup button
* Click Finish Setup in the account tools, you should be redirected to the Stripe KYC
* Edit details still should lead to Stripe Express
* Complete the KYC and get redirected to the overview page
* Finish setup button in Account tools shouldn't be visible

image

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _'QA Testing Not Applicable'_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8597` or branch name `fix/8585-account-tools-finish-setup-button-point-to-kyc-flow` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/8585-account-tools-finish-setup-button-point-to-kyc-flow) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: a57e665e6f76ffda835370a4141841a1cfd099d1
- Build time: 2024-04-10 16:19:45 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/account-status/account-tools/index.tsx [hunk 1]
```diff
@@ -27,7 +29,13 @@ const handleReset = () => {
 };
 
 export const AccountTools: React.FC< Props > = ( props: Props ) => {
-	const accountLink = props.accountLink;
+	let accountLink = props.accountLink;
+	const isPoInProgress = props.poEnabled && ! props.poComplete;
+	if ( isPoInProgress ) {
```
Comment (vladolaru) [inline client/components/account-status/account-tools/index.tsx]: @oaratovskyi As I've shared here (p1712757976717789/1712736009.166459-slack-C03KTTK2YMA), I would keep the `Finish setup` button logic only concerned with initial KYC finished (details_submitted) and not concerned with full KYC for PO. If the initial KYC is completed, we shouldn't show the `Finish setup` button.

Comment (oaratovskyi) [reply client/components/account-status/account-tools/index.tsx]: Ok, sounds good

Comment (oaratovskyi) [reply client/components/account-status/account-tools/index.tsx]: Done!

**Review by vladolaru — APPROVED** [review]
Thank you for working on this and incorporating the changes, Oleks! The testing instructions check out and the code changes look good! :shipit: 

PS: Thank you for also providing automated tests!

**Pull request merged**